File: CodingSamples/Java/Language-Runtime/SubclassTest/demo2/Interval.java
//A Java class that does not explicitly extend another class
//implicitly extends java.lang.Object which is common root
//of Java inheritance model
class Interval {
    
    private int min;
    private int sec;

    public Interval(int m, int s) {
        min = m + s / 60;
        sec = s % 60;
    }

    public int minutes() {
        return min;
    }

    public int seconds() {
        return sec;
    }

    public int time() {
        return 60 * min + sec;
    }

    //overriding toString() method of java.lang.Object
    //to return string representation of this instance
    public String toString() {
        if(sec < 10)
            return min + ":0" + sec;
        return min + ":" + sec;
    }

    //overriding toString() method of java.lang.Object to return
    //an integer which has a different value for instances with
    //different states
    public int hashCode() {
        return min + sec;
    }

    //overriding equals method of java.lang.Object to return true only if the
    //state of this instance matches with the state of other instance
    public boolean equals(Object other) {
        if(other instanceof Interval){
            Interval that = (Interval) other; //explicit down-casting of Object to Interval
            return (min == that.min) && (sec == that.sec);
        }
        return false;
    }

}

File: CodingSamples/DotNet/Language-Runtime/GenericMethodTest/DemoApp2/Interval.cs
using System.Numerics;

class Interval(int min, int sec) : IComparable<Interval>, IAdditionOperators<Interval, Interval, Interval>
{
    public int Minutes { get; } = min + sec / 60;

    public int Seconds { get; } = sec % 60;

    public int Time()
    {
        return 60 * min + sec;
    }

    public static Interval operator+(Interval lhs, Interval rhs)
    {
        return new Interval(lhs.Minutes + rhs.Minutes, lhs.Seconds + rhs.Seconds);
    }

    public override string ToString()
    {
        if(Seconds < 10)
            return Minutes + ":0" + Seconds;
        return Minutes + ":" + Seconds;
    }

    public override int GetHashCode()
    {
        return Minutes + Seconds;
    }

    public override bool Equals(object other)
    {
        if(other is Interval that)
            return (this.Minutes == that.Minutes) && (this.Seconds == that.Seconds);
        return false;
    }

    public int CompareTo(Interval that)
    {
        return this.Time() - that.Time();
    }
}